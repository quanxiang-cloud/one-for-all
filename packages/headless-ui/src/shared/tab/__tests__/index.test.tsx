import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import Tab, { TabNavs } from '../index';

const navs: TabItemProps<React.Key>[] = [
  { id: '1', name: 'tab1', content: 'tab1 content' },
  { id: '2', name: 'tab2', state: 'error', content: 'tab2 content' },
  { id: '3', name: 'tab3', state: 'warning', content: 'tab3 content' },
  { id: '4', name: 'tab4', state: 'error', disabled: true, content: 'tab4 content' },
];

const tabNavsTestProps: TabNavProps<React.Key> = {
  navs,
  style: {
    backgroundColor: 'red',
    width: '80%',
    height: '50px',
  },
  className: 'test-className',
  activeNavClassName: 'test-active-className',
  navsClassName: 'test-navs-className',
  currentKey: '2',
  onClick: (item) => {
    console.log(item);
  },
};

test('tab navs props test', () => {
  const { container } = render(<TabNavs {...tabNavsTestProps} />);
  expect(container).toMatchSnapshot();
});

test('tab navs ref test', () => {
  const tabNavsRef = React.createRef<HTMLDivElement>();
  render(<TabNavs {...tabNavsTestProps} ref={tabNavsRef} />);
  expect(tabNavsRef.current && tabNavsRef.current instanceof HTMLDivElement).toBeTruthy();
});

test('tab navs click test', () => {
  const handleClick = jest.fn();
  const clickTestProps: TabNavProps<React.Key> = {
    navs,
    className: 'tab-click-test-className',
    activeNavClassName: 'tab-click-test-active-className',
    currentKey: '2',
    onClick: handleClick,
  };

  const { container } = render(<TabNavs {...clickTestProps} />);
  fireEvent.click(container.querySelectorAll('.ofa-tab-nav-item')[1]);
  expect(handleClick).toHaveBeenCalledTimes(1);
  expect(container.querySelector('.tab-click-test-active-className')?.innerHTML).toEqual('tab2');
  expect(container).toMatchSnapshot();
});

test('tab props test', () => {
  const tabTestProps: TabsProps<React.Key> = {
    items: navs,
    direction: 'horizon',
    maxHeight: '500px',
    className: 'test-className',
    activeNavClassName: 'test-active-className',
    navsClassName: 'test-navs-className',
    currentKey: '1',
    onChange: (key) => {
      console.log(key);
    },
  };

  const { container } = render(<Tab {...tabTestProps} />);
  expect(container).toMatchSnapshot();
});
