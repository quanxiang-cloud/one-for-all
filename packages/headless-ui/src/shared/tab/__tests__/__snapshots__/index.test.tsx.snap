// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tab navs props test 1`] = `
<div>
  <div
    class="ofa-tab-navs-wrapper"
  >
     
    <div
      class="ofa-tab-navs test-className"
      style="background-color: red; width: 80%; height: 50px;"
    >
      <div
        class="ofa-tab-nav-item test-navs-className"
      >
        tab1
      </div>
      <div
        class="ofa-tab-nav-item tab-nav__error test-navs-className"
      >
        tab2
      </div>
      <div
        class="ofa-tab-nav-item tab-nav__warning test-navs-className"
      >
        tab3
      </div>
      <div
        class="ofa-tab-nav-item ofa-tab-nav-disabled tab-nav__error test-navs-className"
      >
        tab4
      </div>
    </div>
  </div>
</div>
`;

exports[`tab props test 1`] = `
<div>
  <div
    class="ofa-tab-wrapper ofa-tab-direction__horizon test-className"
    style="max-height: 500px;"
  >
    <div
      class="ofa-tab-navs-wrapper"
    >
       
      <div
        class="ofa-tab-navs"
      >
        <div
          class="ofa-tab-nav-item test-navs-className"
        >
          tab1
        </div>
        <div
          class="ofa-tab-nav-item tab-nav__error test-navs-className"
        >
          tab2
        </div>
        <div
          class="ofa-tab-nav-item tab-nav__warning test-navs-className"
        >
          tab3
        </div>
        <div
          class="ofa-tab-nav-item ofa-tab-nav-disabled tab-nav__error test-navs-className"
        >
          tab4
        </div>
      </div>
    </div>
  </div>
</div>
`;
